docs/site/LB3-vs-LB4-request-response-cycle.md

```diff
@@ -0,0 +1,448 @@
+---
+lang: en
+title: 'Differences in LoopBack 3 and LoopBack 4 request/response cycle'
+keywords: LoopBack 4.0, LoopBack 4, LoopBack 3.0, LoopBack 3
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/LB3-vs-LB4-request-response-cycle.html
+---
+
+## Differences between LoopBack 3 and LoopBack 4 request/response cycle
+
+The request/response cycle infrastructure and pathway are very different in
+LoopBack 3 and LoopBack 4. Knowing the differences will help you migrate
+LoopBack 3 apps to LoopBack 4 and implement new request/response related
+features in LoopBack 4.
+
+This document will guide you through the differences and show the LoopBack 4
+equivalent, if there is any.
+
+### Request/response infrastructure
+
+The difference begins with the LoopBack application object itself. In
+LoopBack 3, it is an instance of an Express app; in LoopBack 4, it is not.
+Although LoopBack 4 uses Express as the HTTP server, it is not directly exposed
+anymore.
+
+In LoopBack 3, Express middleware and routers, models, components, boot scripts,
+and remote methods are the ways endpoints can be created on the app. Let's take
+a look at how they have changed and how their functionality can be migrated in
+LoopBack 4.
+
+#### Express middleware and routers
+
+In LoopBack 3 you could add routes and load custom middleware using `app.get()`,
+`app.post()`, `app.use()`, etc., just like how you do in Express.
+In LoopBack 4, you cannot do it yet. However, you can
+[mount a LoopBack 4](https://loopback.io/doc/en/lb4/express-with-lb4-rest-tutorial.html)
+on an Express app, which would allow you to still use the familiar routing
+methods.
+
+{% include tip.html content="Follow our GitHub issues
+[#1293](https://github.com/strongloop/loopback-next/issues/1293)
+and
+[#2035](https://github.com/strongloop/loopback-next/issues/2035)
+to track the progress on supporting Express middlweware in LoopBack 4." %}
+
+If you want to mount a Express router, you can use the
+[RestApplication.mountExpressRouter()](https://loopback.io/doc/en/lb4/apidocs.rest.restapplication.mountexpressrouter.html)
+API.
+
+Using [Controllers](https://loopback.io/doc/en/lb4/Controllers.html) is the
+recommended way for creating custom (and REST) endpoints on your app. Its
+support for
+[dependency injection](https://loopback.io/doc/en/lb4/Dependency-injection.html)
+and [Interceptors](https://loopback.io/doc/en/lb4/Interceptors.html) makes it a
+very powerful extension mechanism.
```

Comment:
Maybe we can add some comments (or examples) to show the differences of using controllers in LB4 and why we are using it.

Reply:
A link to the controllers doc should be sufficient. If we get into all the details, it will dilute theme of the doc.